From 327eabfd36975d76864370670eb4c7a3377ff10b Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <b02247@freescale.com>
Date: Tue, 5 Nov 2013 13:34:58 +0800
Subject: [PATCH 41/50] ENGR00276692 Error when turn off BT in audio playback

This issue is caused by the lock isn't released in getFlags().
The getOutput will be blocked, then the Music player will no response.

The issue is introduced by 971d08e0985c4f952213406531512c9b3b473070

Signed-off-by: Shengjiu Wang <b02247@freescale.com>
---
 media/libmedia/AudioSystem.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/libmedia/AudioSystem.cpp b/media/libmedia/AudioSystem.cpp
index ac380d5..2c3d09e 100644
--- a/media/libmedia/AudioSystem.cpp
+++ b/media/libmedia/AudioSystem.cpp
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2006-2007 The Android Open Source Project
-** Copyright (C) 2012 Freescale Semiconductor, Inc.
+ * Copyright (C) 2012-2013 Freescale Semiconductor, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -308,8 +308,8 @@ status_t AudioSystem::getFlags(audio_io_handle_t output,
     outputDesc = AudioSystem::gOutputs.valueFor(output);
     if (outputDesc != NULL) {
         *flags = outputDesc->flags;
-        gLock.unlock();
     }
+    gLock.unlock();
 
     ALOGV("getFlags() streamType %d, output %d, flags %d", streamType, output, *flags);
 
-- 
1.8.0

